recipe/meta.yaml

```diff
@@ -152,7 +152,8 @@ outputs:
         - {{ pin_subpackage('ray-core', exact=True) }}
         - numpy >=1.20
         - pandas >=1.3
-        - pyarrow >=6.0.1
+        - pyarrow >=6.0.1  # [not win]
+        - pyarrow >=6.0.1,<7.0.0  # [win]
```

Comment:
This is an ancient arrow version and not supported anymore in conda-forge (receives no more migrations etc.). I'm opposed to this cap, even if it's used upstream. Do we have an idea what the specific was?

Reply:
There is a comment in the original setup.py 
> Serialization workaround for pyarrow 7.0.0+ doesn't work for Windows.

This came from ray-project/ray#29993 which fixes ray-project/ray#29814 which opens with this comment
> When pickling Arrow data (tables, in our case), Arrow currently doesn't truncate the data buffers for zero-copy slice views of the data (see [Arrow ticket](https://issues.apache.org/jira/browse/ARROW-10739)). For many of our workloads, this results is sending a potentially very large data buffer over the network for a very small slice. For pathological cases like shuffling, where we might chunk a data block into 1000 chunks and send each to a different reduce, this would involve sending 1000 copies of the entire data block; for a 1 GiB data block, we'd send an aggregate 1 TiB of data over the network instead of the expected 1 GiB. This makes Ray + Arrow essentially unusable.

> We can bypass this buggy pickle path by registering a custom serializer for Arrow data that properly truncates these buffers.

Then the arrow issue tracker migrated to github, and the upstream issue became apache/arrow#26685. The last action was to remove it from the 11.0 milestone.

So as I understand the current status: there is a workaround for this arrow issue that works on non-windows, but fails for `pyarrow >= 7.0` on windows.